src/datapilot/core/platforms/dbt/insights/checks/check_column_name_contract.py

```diff
@@ -0,0 +1,94 @@
+    )
+    REASON_TO_FLAG = (
+        "Column names that do not adhere to the contract can lead to confusion and hinder effective data "
+        "modeling and analysis. It's important to maintain consistent column naming conventions."
+    )
+    FAILURE_MESSAGE = (
+        "The following columns in the model `{model_unique_id}` do not adhere to the contract:\n{columns}. "
+        "Inconsistent column naming conventions can impede understanding and usage of the model."
+    )
+    RECOMMENDATION = (
+        "Update the column names listed above in the model `{model_unique_id}` to adhere to the contract. "
+        "Consistent column naming conventions provide valuable context and aids in data understanding and collaboration."
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.pattern = get_contract_regex_configuration(self.config)
+        self.dtypes = get_dtypes_configuration(self.config)
+
+    def generate(self, *args, **kwargs) -> List[DBTModelInsightResponse]:
+        insights = []
+        for node_id, node in self.nodes.items():
+            if self.should_skip_model(node_id):
+                self.logger.debug(f"Skipping model {node_id} as it is not enabled for selected models")
+                continue
+            if node.resource_type == AltimateResourceType.model:
+                columns = self._get_columns_with_contract_violation(node_id)
+                if columns:
+                    insights.append(
+                        DBTModelInsightResponse(
+                            unique_id=node_id,
+                            package_name=node.package_name,
+                            path=node.original_file_path,
+                            original_file_path=node.original_file_path,
+                            insight=self._build_failure_result(node_id, columns),
+                            severity=get_severity(self.config, self.ALIAS, self.DEFAULT_SEVERITY),
+                        )
+                    )
+        return insights
+
+    def _build_failure_result(self, model_unique_id: str, columns: Sequence[str]) -> DBTInsightResult:
+        failure_message = self.FAILURE_MESSAGE.format(
+            columns=numbered_list(columns),
+            model_unique_id=model_unique_id,
+        )
+        recommendation = self.RECOMMENDATION.format(model_unique_id=model_unique_id)
+
+        return DBTInsightResult(
+            type=self.TYPE,
+            name=self.NAME,
+            message=failure_message,
+            recommendation=recommendation,
+            reason_to_flag=self.REASON_TO_FLAG,
+            metadata={"columns": columns, "model_unique_id": model_unique_id},
+        )
+
+    def _get_columns_with_contract_violation(self, node_id) -> Sequence[str]:
+        columns_with_contract_violation = []
+        for col_name, col_node in self.get_node(node_id).columns.items():
```

Comment:
This will get columns from manifest. manifest will have less columns than catalog. So we need to fetch those

Reply:
set.catalog_wrapper.get_schemas or sometibng willl take car eiof ut